app/src/main/java/com/example/plantsapp/presentation/PlantApplication.kt

```diff
@@ -1,18 +1,68 @@
 package com.example.plantsapp.presentation
 
 import android.app.Application
+import android.content.Context
+import androidx.hilt.work.HiltWorkerFactory
+import androidx.work.Configuration
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import com.bumptech.glide.Glide
 import com.example.plantsapp.BuildConfig
+import com.example.plantsapp.presentation.ui.notification.NotificationWorker
+import com.example.plantsapp.presentation.ui.utils.calculateDelay
 import dagger.hilt.android.HiltAndroidApp
 import timber.log.Timber
+import java.util.Calendar
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 
 @HiltAndroidApp
-class PlantApplication : Application() {
+class PlantApplication : Application(), Configuration.Provider {
+
+    @Inject
+    lateinit var workerFactory: HiltWorkerFactory
 
     override fun onCreate() {
         super.onCreate()
 
         if (BuildConfig.DEBUG) {
             Timber.plant(Timber.DebugTree())
+
+            Thread {
+                Glide.get(this).clearDiskCache()
+            }.start()
```

Comment:
I don't think this is a good idea to start a thread here - and clearDiscCach
 Lets remove it - and instead add   .diskCacheStrategy(DiskCacheStrategy.NONE) to glide requests 
BUT - it is out of scope for this task , so I've created additional task for this + refactoring
https://github.com/Lonexera/students-lab/projects/1#card-73123534


For now just remove this